jwql/instrument_monitors/common_monitors/dark_monitor.py

```diff
@@ -0,0 +1,913 @@
+from jwql.database.database_interface import MIRIDarkQueries, MIRIDarkPixelStats, MIRIDarkDarkCurrent
+from jwql.database.database_interface import NIRSpecDarkQueries, NIRSpecDarkPixelStats, NIRSpecDarkDarkCurrent
+from jwql.database.database_interface import FGSDarkQueries, FGSDarkPixelStats, FGSDarkDarkCurrent
+from jwql.instrument_monitors import pipeline_tools
+from jwql.jwql_monitors import monitor_mast
+from jwql.utils import maths, instrument_properties, permissions
+from jwql.utils.constants import AMPLIFIER_BOUNDARIES, JWST_INSTRUMENT_NAMES, \
+                                 JWST_INSTRUMENT_NAMES_MIXEDCASE, JWST_DATAPRODUCTS
+from jwql.utils.logging_functions import configure_logging, log_info, log_fail
+from jwql.utils.permissions import set_permissions
+from jwql.utils.utils import copy_files, download_mast_data, ensure_dir_exists, get_config, \
+                             filesystem_path
+
+THRESHOLDS_FILE = os.path.join(os.path.split(__file__)[0], 'dark_monitor_file_thresholds.txt')
+
+
+def mast_query_darks(instrument, aperture, start_date, end_date):
+    """Use astroquery to search MAST for dark current data
+
+    Parameters
+    ----------
+    instrument : str
+        Instrument name
+
+    aperture : str
+        Detector aperture to search for (e.g. NRCA1_FULL)
+
+    start_date : float
+        Starting date for the search in MJD
+
+    end_date : float
+        Ending date for the search in MJD
+
+    Returns
+    -------
+    query_results : list
+        List of dictionaries containing the query results
+    """
+    # Make sure instrument is correct case
+    if instrument.lower() == 'nircam':
+        instrument = 'NIRCam'
+        dark_template = ['NRC_DARK']
+    elif instrument.lower() == 'niriss':
+        instrument = 'NIRISS'
+        dark_template = ['NIS_DARK']
+    elif instrument.lower() == 'nirspec':
+        instrument = 'NIRSpec'
+        dark_template = 'NRS_DARK'
+    elif instrument.lower() == 'fgs':
+        instrument = 'FGS'
+        dark_template = 'FGS_DARK'
+    elif instrument.lower() == 'miri':
+        instrument = 'MIRI'
+        dark_template = ['MIR_DARKALL', 'MIR_DARKIMG', 'MIR_DARKMRS']
+
+    # monitor_mast.instrument_inventory does not allow list inputs to
+    # the added_filters input (or at least if you do provide a list, then
+    # it becomes a nested list when it sends the query to MAST. The
+    # nested list is subsequently ignored by MAST.)
+    # So query once for each dark template, and combine outputs into a
+    # single list.
+    query_results = []
+    for template_name in dark_template:
+        # Create dictionary of parameters to add
+        parameters = {"date_obs_mjd": {"min": start_date, "max": end_date},
+                      "apername": aperture, "exp_type": template_name}
+
+        query = monitor_mast.instrument_inventory(instrument, dataproduct=JWST_DATAPRODUCTS,
```

Comment:
@bhilbert4 Nice use of our own software!

Reply:
Recycling!